source/s3_client.c

```diff
@@ -48,6 +48,46 @@ PyObject *aws_py_s3_get_recommended_throughput_target_gbps(PyObject *self, PyObj
     return PyFloat_FromDouble(platform_info->max_throughput_gbps);
 }
 
+PyObject *aws_py_s3_get_optimized_platforms(PyObject *self, PyObject *args) {
+    (void)self;
+    (void)args;
+
+    bool success = false;
+    struct aws_array_list platform_list = aws_s3_get_platforms_with_recommended_config();
+
+    size_t list_length = aws_array_list_length(&platform_list);
+
+    PyObject *py_list = PyList_New(list_length);
+    if (!py_list) {
+        PyErr_SetString(PyExc_MemoryError, "Failed to allocate memory for platform list");
```

Comment:
you don't need to set an exception. PyXYZ() functions will have already set one if they fail
```suggestion
```